model/build-processor/src/main/java/org/keycloak/models/map/processor/AbstractGenerateEntityImplementationsProcessor.java

```diff
@@ -215,6 +215,25 @@ protected String deepClone(TypeMirror fieldType, String parameterName) {
         return "deepClone(" + parameterName + ")";
     }
 
+    protected String removeUndefined(TypeMirror fieldType, String parameterName) {
+        TypeElement typeElement = elements.getTypeElement(types.erasure(fieldType).toString());
+        boolean isMapType = isMapType(typeElement);
+        TypeMirror collectionItemTypeMirror = getGenericsDeclaration(fieldType).get(isMapType ? 1 : 0);
+
+        return parameterName + (isMapType ? ".values()" : "") + ".removeIf(item -> " + isUndefined(collectionItemTypeMirror, "item") + ")";
+    }
+
+    protected String isUndefined(TypeMirror fieldType, String parameterName) {
+        TypeElement typeElement = elements.getTypeElement(types.erasure(fieldType).toString());
+        return parameterName  + " == null" + (
+                isCollection(fieldType)
+                ? " || " + parameterName + ".isEmpty()" + (
+                        !isMapType(typeElement)
+                        ? " || " + parameterName + ".stream().allMatch(e -> " + isUndefined(getGenericsDeclaration(fieldType).get(0), "e") + ")"
+                        : "")
+                : "");
```

Comment:
This should be optimized to leverage `Objects::isNull` reference rather than creating many `(e -> e == null)` functions when the field type is not a collection.

Reply:
It would be possible. We can even do it similarly as we do it within HotRod. We can have some util class with overloaded `isUndefined` method. This way we could even specify class specific undefined conditions e.g. `"".equals(stringParameter.trim())`.